pubsub/subscriptions.rb

```diff
@@ -0,0 +1,284 @@
+end
+
+def list_subscriptions project_id:
+  # [START list_subscriptions]
+  # project_id = Your Google Cloud Project ID
+  require "google/cloud/pubsub"
+
+  pubsub        = Google::Cloud::Pubsub.new project: project_id
+
+  subscriptions = pubsub.list_subscriptions
+
+  puts "Subscriptions:"
+  subscriptions.each do |subscription|
+    puts subscription.name
+  end
+  # [END list_subscriptions]
+end
+
+def delete_subscription project_id:, subscription_name:
+  # [START delete_subscription]
+  # project_id        = "Your Google Cloud Project ID"
+  # subscription_name = "Your Pubsub subscription name"
+  require "google/cloud/pubsub"
+
+  pubsub       = Google::Cloud::Pubsub.new project: project_id
+
+  subscription = pubsub.subscription subscription_name
+  subscription.delete
+
+  puts "Subscription #{subscription_name} deleted."
+  # [END delete_subscription]
+end
+
+def get_subscription_policy project_id:, subscription_name:
+  # [START get_subscription_policy]
+  # project_id        = "Your Google Cloud Project ID"
+  # subscription_name = "Your Pubsub subscription name"
+  require "google/cloud/pubsub"
+
+  pubsub       = Google::Cloud::Pubsub.new project: project_id
+
+  subscription = pubsub.subscription subscription_name
+  policy       = subscription.policy
+
+  puts "Subscription policy:"
+  puts policy.roles
+  # [END get_subscription_policy]
+end
+
+def set_subscription_policy project_id:, subscription_name:
+  # [START set_subscription_policy]
+  # project_id        = "Your Google Cloud Project ID"
+  # subscription_name = "Your Pubsub subscription name"
+  require "google/cloud/pubsub"
+
+  pubsub       = Google::Cloud::Pubsub.new project: project_id
+
+  subscription = pubsub.subscription subscription_name
+  subscription.policy do |policy|
+    policy.add "roles/pubsub.subscriber",
+      "serviceAccount:account-name@project-name.iam.gserviceaccount.com"
+  end
+  # [END set_subscription_policy]
+end
+
+def test_subscription_permissions project_id:, subscription_name:
+  # [START test_subscription_permissions]
+  # project_id        = "Your Google Cloud Project ID"
+  # subscription_name = "Your Pubsub subscription name"
+  require "google/cloud/pubsub"
+
+  pubsub       = Google::Cloud::Pubsub.new project: project_id
+
+  subscription = pubsub.subscription subscription_name
+  permissions  = subscription.test_permissions "pubsub.subscriptions.consume",
+    "pubsub.subscriptions.update"
+
+  puts "Permission to consume" if permissions.include? "pubsub.subscriptions.consume"
+  puts "Permission to update" if permissions.include? "pubsub.subscriptions.update"
+  # [END test_subscription_permissions]
+end
+
+def listen_for_messages project_id:, subscription_name:
+  # [START listen_for_messages]
+  # project_id        = "Your Google Cloud Project ID"
+  # subscription_name = "Your Pubsub subscription name"
+  require "google/cloud/pubsub"
+
+  pubsub       = Google::Cloud::Pubsub.new project: project_id
```

Comment:
This line isn't part of a group of assignments could you remove the added space between the variable and the assignment operator?
`pubsub = Google::Cloud::Pubsub.new project: project_id`